### PR TITLE
Add Scala installation check when new Scala project and minor bug fix

### PR DESCRIPTION
**Pull request by djlan**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			@Override
+			public void run() {
+				final String jarPathString = sparkLibraryOptionsPanel.getSparkLibraryPath();
+				if (StringUtils.isNullOrEmpty(jarPathString)) {
+					DefaultLoader.getUIHelper().showError("Spark Library Path cannot be null",
+							"Spark Project Settings");
+				} else {
+					IPath jarPath = new Path(jarPathString);
+					IClasspathEntry sparkEntry = JavaCore.newLibraryEntry(jarPath, null, null);
+					System.arraycopy(entries, 0, newEntries, 0, entries.length);
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java]: Why we need array copy here?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			throws CoreException, InterruptedException {
+		if (monitor == null) {
+			monitor = new NullProgressMonitor();
+		}
+		byte nSteps = 6;
+		((IProgressMonitor) monitor).beginTask(NewWizardMessages.JavaCapabilityConfigurationPage_op_desc_java,
+				nSteps);
+		try {
+			IProject project = addHDInsightNature(monitor);
+			Method method = JavaCapabilityConfigurationPage.class.getDeclaredMethod("getBuildPathsBlock");
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java]: We call a private method here? Why?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java [hunk 3]
```diff
@@ -19,71 +19,66 @@
...
 		page2.setTitle("HDInsight Spark Project Library Settings");
 		page2.setDescription("Define the project build settings.");
 	}
+	
+	private static boolean setFocusToInstallationWindow() {
+		// Unfortunately, the marketplace client windows can not get the focus so do the trick here since the parent is the project wizard
+		if (PluginUtil.checkPlugInInstallation(PluginUtil.scalaPluginSymbolicName) != true) {
+			Shell parent = PluginUtil.getParentShell();
+			
+			int isModeless = parent.getStyle() & SWT.MODELESS;
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java]: Is it a bit AND operation?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java [hunk 4]
```diff
@@ -19,71 +19,66 @@
...
+		if (PluginUtil.checkPlugInInstallation(PluginUtil.scalaPluginSymbolicName) != true) {
+			Shell parent = PluginUtil.getParentShell();
+			
+			int isModeless = parent.getStyle() & SWT.MODELESS; 
+			if (isModeless == 0) {
+				parent.close();
+			}
+		}
+		
+		return true;
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java]: Looks like always return true.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java (hunk 1 above)
Comment (voidfunction) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java]: those code are copied from HDInsightScalaProjectWizard. We are willing to return a new entries with spark library support and keep the origin entries of the super class. 

---
In reply to: [132864101](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132864101) [](ancestors = 132864101)

Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageOne.java]: Yes


---
In reply to: [132867205](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132867205) [](ancestors = 132867205,132864101)

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java (hunk 2 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java]: Actually it is just moved from the wizard to here.

---
In reply to: [132864164](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132864164) [](ancestors = 132864164)

Comment (voidfunction) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java]: we need to trigger configureJavaProject but we didn't have a better way to do this. So we just reference what Scala IDE do.

---
In reply to: [132867561](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132867561) [](ancestors = 132867561,132864164)

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/PluginUtil.java [hunk 5]
```diff
@@ -86,6 +86,9 @@
 public class PluginUtil {
 
 	public static final String pluginFolder = getPluginFolderPathUsingBundle();
+
+	public static String scalaPluginMarketplaceURL = "http://marketplace.eclipse.org/marketplace-client-intro?mpc_install=421";
+	public static String scalaPluginSymbolicName = "org.scala-ide.sdt.core";
```
Comment (voidfunction) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/PluginUtil.java]: final?

**Comment by voidfunction** [comment]
:shipit:

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java (hunk 4 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java]: Discussed, this is to workaround the constructor issue.

---
In reply to: [132864276](https://github.com/Microsoft/azure-tools-for-java/pull/884#discussion_r132864276) [](ancestors = 132864276)

**Pull request merged**